diff --git a/src/findlib/Makefile b/src/findlib/Makefile
index b2317c1..e99e412 100644
--- a/src/findlib/Makefile
+++ b/src/findlib/Makefile
@@ -42,10 +42,10 @@ DYNLOAD_OBJECTS  = fl_dynload.cmo
 DYNLOAD_XOBJECTS = $(DYNLOAD_OBJECTS:.cmo=.cmx)
 
 
-all: ocamlfind$(EXEC_SUFFIX) findlib.cma findlib_top.cma topfind $(NUMTOP) \
+all: findlib.cma findlib_top.cma topfind $(NUMTOP) \
      findlib_dynload.cma
 
-opt: ocamlfind_opt$(EXEC_SUFFIX) findlib.cmxa findlib_top.cmxa topfind \
+opt: findlib.cmxa findlib_top.cmxa topfind \
      findlib_dynload.cmxa
 
 num-top: num_top.cma
@@ -145,8 +145,6 @@ install: all
 	findlib_dynload.cma findlib_dynload.cmxa findlib_dynload$(LIB_SUFFIX) findlib_dynload.cmxs fl_dynload.mli fl_dynload.cmi \
 	META` && \
 	$(CP) $$files "$(DESTDIR)$(prefix)$(OCAML_SITELIB)/$(NAME)/"
-	f="ocamlfind$(EXEC_SUFFIX)"; { test -f ocamlfind_opt$(EXEC_SUFFIX) && f="ocamlfind_opt$(EXEC_SUFFIX)"; }; \
-	$(INSTALLFILE) $$f "$(DESTDIR)$(prefix)$(OCAMLFIND_BIN)/ocamlfind$(EXEC_SUFFIX)"
 	# the following "if" block is only needed for 4.00beta2
 	if [ $(OCAML_REMOVE_DIRECTORY) -eq 0 -a -f "$(OCAML_CORE_STDLIB)/compiler-libs/topdirs.cmi" ]; then \
 	    cd "$(OCAML_CORE_STDLIB)/compiler-libs/"; \
