diff --git a/src/findlib/Makefile b/src/findlib/Makefile
index b2317c1..50ec7f5 100644
--- a/src/findlib/Makefile
+++ b/src/findlib/Makefile
@@ -42,10 +42,10 @@ DYNLOAD_OBJECTS  = fl_dynload.cmo
 DYNLOAD_XOBJECTS = $(DYNLOAD_OBJECTS:.cmo=.cmx)
 
 
-all: ocamlfind$(EXEC_SUFFIX) findlib.cma findlib_top.cma topfind $(NUMTOP) \
+all: findlib.cma findlib_top.cma topfind $(NUMTOP) \
      findlib_dynload.cma
 
-opt: ocamlfind_opt$(EXEC_SUFFIX) findlib.cmxa findlib_top.cmxa topfind \
+opt: findlib.cmxa findlib_top.cmxa topfind \
      findlib_dynload.cmxa
 
 num-top: num_top.cma
