diff --git a/otherlibs/win32unix/open.c b/otherlibs/win32unix/open.c
index 5a56d92dc5..c324ce0ea2 100644
--- a/otherlibs/win32unix/open.c
+++ b/otherlibs/win32unix/open.c
@@ -24,7 +24,7 @@
 
 static int open_access_flags[15] = {
   GENERIC_READ, GENERIC_WRITE, GENERIC_READ|GENERIC_WRITE,
-  0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+  0, FILE_APPEND_DATA, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
 };
 
 static int open_create_flags[15] = {
@@ -46,6 +46,7 @@ CAMLprim value unix_open(value path, value flags, value perm)
   int fileaccess, createflags, fileattrib, filecreate, sharemode, cloexec;
   SECURITY_ATTRIBUTES attr;
   HANDLE h;
+  DWORD dwMoved;
   wchar_t * wpath;
 
   caml_unix_check_path(path, "open");
@@ -87,5 +88,15 @@ CAMLprim value unix_open(value path, value flags, value perm)
     win32_maperr(GetLastError());
     uerror("open", path);
   }
+
+  if (fileaccess & FILE_APPEND_DATA) {
+    dwMoved = SetFilePointer(h, 0, NULL, FILE_END);
+    if (dwMoved == INVALID_SET_FILE_POINTER) {
+      caml_win32_maperr(GetLastError());
+      CloseHandle(h);
+      caml_uerror("open", path);
+    }
+  }
+
   return win_alloc_handle(h);
 }
