--- ocaml-4.14.0.old/configure	2022-03-28 12:27:36.000000000 +0000
+++ ocaml-4.14.0/configure	2023-01-29 23:35:10.270519014 +0000
@@ -12657,7 +12657,7 @@
 # Libraries to build depending on the host

 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     unix_or_win32="win32"
     unixlib="win32unix"
      ;; #(
@@ -12666,7 +12666,7 @@
   unixlib="unix" ;;
 esac
 case $host in #(
-  *-*-cygwin*|*-*-mingw32|*-pc-windows) :
+  *-*-cygwin*|*-*-mingw32*|*-pc-windows) :
     exeext=".exe" ;; #(
   *) :
     exeext='' ;;
@@ -12796,7 +12796,7 @@
 # the vendor only as a fall-back. We could put tis part of the logic
 # in the macro itself, too
 case $host in #(
-  *-*-mingw32) :
+  *-*-mingw32*) :
     case $ocaml_cv_cc_vendor in #(
   gcc-[01234]-*) :
     as_fn_error $? "This version of Mingw GCC is too old. Please use GCC version 5 or above." "$LINENO" 5 ;; #(
@@ -12889,7 +12889,7 @@

 # Enable SSE2 on x86 mingw to avoid using 80-bit registers.
 case $host in #(
-  i686-*-mingw32) :
+  i686-*-mingw32*) :
     internal_cflags="$internal_cflags -mfpmath=sse -msse2" ;; #(
   *) :
      ;;
@@ -12916,7 +12916,7 @@
 if test x"$enable_shared" = "xno"; then :
   supports_shared_libraries=false
   case $host in #(
-  *-pc-windows|*-w64-mingw32) :
+  *-pc-windows|*-w64-mingw32*) :
     as_fn_error $? "Cannot build native Win32 with --disable-shared" "$LINENO" 5 ;; #(
   *) :
      ;;
@@ -12935,10 +12935,10 @@
     flexlink_flags="-chain $flexdll_chain -merge-manifest -stack 16777216" ;; #(
   *-*-cygwin*) :
     as_fn_error $? "unknown cygwin variant" "$LINENO" 5 ;; #(
-  i686-w64-mingw32) :
+  i686-w64-mingw32*) :
     flexdll_chain='mingw'
     flexlink_flags="-chain $flexdll_chain -stack 16777216" ;; #(
-  x86_64-w64-mingw32) :
+  x86_64-w64-mingw32*) :
     flexdll_chain='mingw64'
     flexlink_flags="-chain $flexdll_chain -stack 33554432" ;; #(
   i686-pc-windows) :
@@ -12962,7 +12962,7 @@
 else
   flexmsg=''
     case $target in #(
-  *-*-cygwin*|*-w64-mingw32|*-pc-windows) :
+  *-*-cygwin*|*-w64-mingw32*|*-pc-windows) :
     if test x"$with_flexdll" = 'x' -o x"$with_flexdll" = 'xflexdll'; then :
   if test -f 'flexdll/flexdll.h'; then :
   flexdir=flexdll
@@ -13118,7 +13118,7 @@


     case $host in #(
-  *-w64-mingw32|*-pc-windows) :
+  *-pc-windows) :
     flexlink_where="$(cmd /c "$flexlink" -where 2>/dev/null)"
       if test -z "$flexlink_where"; then :
   as_fn_error $? "$flexlink is not executable from a native Win32 process" "$LINENO" 5
@@ -13234,7 +13234,7 @@
     supports_shared_libraries=false
    { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: flexdll.h not found: shared library support disabled." >&5
 $as_echo "$as_me: WARNING: flexdll.h not found: shared library support disabled." >&2;} ;; #(
-  no,*,*-w64-mingw32|no,*,*-pc-windows) :
+  no,*,*-w64-mingw32*|no,*,*-pc-windows) :
     as_fn_error $? "flexdll.h is required for native Win32" "$LINENO" 5 ;; #(
   *) :
      ;;
@@ -13245,7 +13245,7 @@
     supports_shared_libraries=false
     { $as_echo "$as_me:${as_lineno-$LINENO}: WARNING: flexlink not found: shared library support disabled." >&5
 $as_echo "$as_me: WARNING: flexlink not found: shared library support disabled." >&2;} ;; #(
-  ,*,,*-w64-mingw32|,*,,*-pc-windows) :
+  ,*,,*-w64-mingw32*|,*,,*-pc-windows) :
     as_fn_error $? "flexlink is required for native Win32" "$LINENO" 5 ;; #(
   *) :
      ;;
@@ -13269,7 +13269,7 @@

 fi
     ostype="Cygwin" ;; #(
-  *,*-*-mingw32) :
+  *,*-*-mingw32*) :
     case $host in #(
   i686-*-*) :
     oc_dll_ldflags="-static-libgcc" ;; #(
@@ -14090,7 +14090,7 @@
                    -flat_namespace -undefined suppress -Wl,-no_compact_unwind \
                    \$(LDFLAGS)"
       supports_shared_libraries=true ;; #(
-  *-*-mingw32) :
+  *-*-mingw32*) :
     mksharedlib='$(FLEXLINK)'
       mkmaindll='$(FLEXLINK) -maindll'
       if test -n "$oc_dll_ldflags"; then :
@@ -14153,7 +14153,7 @@
   case "$host" in #(
   *-*-cygwin*) :
     natdynlink=true ;; #(
-  *-*-mingw32) :
+  *-*-mingw32*) :
     natdynlink=true ;; #(
   *-pc-windows) :
     natdynlink=true ;; #(
@@ -14317,7 +14317,7 @@
     arch=i386; system=cygwin ;; #(
   i[3456]86-*-gnu*) :
     arch=i386; system=gnu ;; #(
-  i[3456]86-*-mingw32) :
+  i[3456]86-*-mingw32*) :
     arch=i386; system=mingw ;; #(
   i686-pc-windows) :
     arch=i386; system=win32 ;; #(
@@ -14388,7 +14388,7 @@
     arch=arm64; system=macosx ;; #(
   x86_64-*-darwin*) :
     arch=amd64; system=macosx ;; #(
-  x86_64-*-mingw32) :
+  x86_64-*-mingw32*) :
     arch=amd64; system=mingw64 ;; #(
   aarch64-*-linux*) :
     arch=arm64; system=linux ;; #(
@@ -14756,7 +14756,7 @@
 fi
   if test "$cross_compiling" = yes; then :
   case $target in #(
-  x86_64-w64-mingw32) :
+  x86_64-w64-mingw32*) :
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling; assume not" >&5
 $as_echo "cross-compiling; assume not" >&6; } ;; #(
   *) :
@@ -14789,7 +14789,7 @@
     hard_error=true ;; #(
   yes,*) :
     hard_error=false ;; #(
-  *,x86_64-w64-mingw32) :
+  *,x86_64-w64-mingw32*) :
     hard_error=false ;; #(
   *) :
     hard_error=true ;;
@@ -14820,7 +14820,7 @@
 fi
   if test "$cross_compiling" = yes; then :
   case $target in #(
-  x86_64-w64-mingw32|x86_64-*-cygwin*) :
+  x86_64-w64-mingw32*|x86_64-*-cygwin*) :
     { $as_echo "$as_me:${as_lineno-$LINENO}: result: cross-compiling; assume not" >&5
 $as_echo "cross-compiling; assume not" >&6; } ;; #(
   *) :
@@ -14875,7 +14875,7 @@
     hard_error=true ;; #(
   yes,*) :
     hard_error=false ;; #(
-  *,x86_64-w64-mingw32|*,x86_64-*-cygwin*) :
+  *,x86_64-w64-mingw32*|*,x86_64-*-cygwin*) :
     hard_error=false ;; #(
   *) :
     case $ocaml_cv_cc_vendor in #(
@@ -15159,7 +15159,7 @@
 sockets=true

 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     cclibs="$cclibs -lws2_32"
     { $as_echo "$as_me:${as_lineno-$LINENO}: checking for library containing socket" >&5
 $as_echo_n "checking for library containing socket... " >&6; }
@@ -15423,7 +15423,7 @@
 ## socklen_t

 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     ac_fn_c_check_type "$LINENO" "socklen_t" "ac_cv_type_socklen_t" "#include <ws2tcpip.h>
 "
 if test "x$ac_cv_type_socklen_t" = xyes; then :
@@ -15451,7 +15451,7 @@
 ## Unix domain sockets support on Windows

 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     for ac_header in afunix.h
 do :
   ac_fn_c_check_header_compile "$LINENO" "afunix.h" "ac_cv_header_afunix_h" "#include <winsock2.h>
@@ -15475,7 +15475,7 @@
 ipv6=true

 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     ac_fn_c_check_type "$LINENO" "struct sockaddr_in6" "ac_cv_type_struct_sockaddr_in6" "#include <ws2tcpip.h>
 "
 if test "x$ac_cv_type_struct_sockaddr_in6" = xyes; then :
@@ -15581,7 +15581,7 @@
 # seem to detect it properly on Windows so we hardcode the definition
 # of HAS_UTIME on Windows but this will probably need to be clarified
 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     $as_echo "#define HAS_UTIME 1" >>confdefs.h
  ;; #(
   *) :
@@ -15775,7 +15775,7 @@
 # Note: detection fails on Windows so hardcoding the result
 # (should be debugged later)
 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     $as_echo "#define HAS_GETHOSTNAME 1" >>confdefs.h
  ;; #(
   *) :
@@ -15827,7 +15827,7 @@
 ## setsid

 case $host in #(
-  *-cygwin|*-*-mingw32|*-pc-windows) :
+  *-cygwin|*-*-mingw32*|*-pc-windows) :
      ;; #(
   *) :
     ac_fn_c_check_func "$LINENO" "setsid" "ac_cv_func_setsid"
@@ -15928,7 +15928,7 @@
 ## shared library support
 if $supports_shared_libraries; then :
   case $host in #(
-  *-*-mingw32|*-pc-windows|*-*-cygwin*) :
+  *-*-mingw32*|*-pc-windows|*-*-cygwin*) :
     DLLIBS="" ;; #(
   *) :
     ac_fn_c_check_func "$LINENO" "dlopen" "ac_cv_func_dlopen"
@@ -16023,7 +16023,7 @@

 ## -fdebug-prefix-map support by the C compiler
 case $ocaml_cv_cc_vendor,$host in #(
-  *,*-*-mingw32) :
+  *,*-*-mingw32*) :
     cc_has_debug_prefix_map=false ;; #(
   *,*-pc-windows) :
     cc_has_debug_prefix_map=false ;; #(
@@ -16677,7 +16677,7 @@
 $as_echo "$as_me: the Win32/POSIX threads library is disabled" >&6;} ;; #(
   *) :
     case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     systhread_support=true
       otherlibraries="$otherlibraries systhreads"
       { $as_echo "$as_me:${as_lineno-$LINENO}: the Win32 threads library is supported" >&5
@@ -17374,7 +17374,7 @@
 asm_cfi_supported=false
 if $native_compiler; then :
   case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
      ;; #(
   *) :

@@ -17822,7 +17822,7 @@
 cclibs="$cclibs $mathlib"

 case $host in #(
-  *-*-mingw32) :
+  *-*-mingw32*) :
     bytecclibs="-lws2_32 -lversion"
     nativecclibs="-lws2_32 -lversion" ;; #(
   *-pc-windows) :
@@ -17842,7 +17842,7 @@
 fi

 case $host in #(
-  *-*-mingw32|*-pc-windows) :
+  *-*-mingw32*|*-pc-windows) :
     case $WINDOWS_UNICODE_MODE in #(
   ansi) :
     windows_unicode=0 ;; #(
@@ -17858,9 +17858,9 @@
 # Define default prefix correctly for the different Windows ports
 if test x"$prefix" = "xNONE"; then :
   case $host in #(
-  i686-w64-mingw32) :
+  i686-w64-mingw32*) :
     prefix='C:/ocamlmgw' ;; #(
-  x86_64-w64-mingw32) :
+  x86_64-w64-mingw32*) :
     prefix='C:/ocamlmgw64' ;; #(
   i686-pc-windows) :
     prefix='C:/ocamlms' ;; #(
@@ -17885,7 +17885,7 @@
 # but whose value is not guessed properly by configure
 # (all this should be understood and fixed)
 case $host in #(
-  *-*-mingw32) :
+  *-*-mingw32*) :
     $as_echo "#define HAS_BROKEN_PRINTF 1" >>confdefs.h

     $as_echo "#define HAS_STRERROR 1" >>confdefs.h
