diff --git a/src/findlib/Makefile b/src/findlib/Makefile
index 6ec8a71..d3e79eb 100644
--- a/src/findlib/Makefile
+++ b/src/findlib/Makefile
@@ -13,9 +13,9 @@ include $(TOP)/Makefile.config
 NAME = findlib

 # Need compiler-libs since ocaml-4.00
-OCAMLC = ocamlc -I +compiler-libs
-OCAMLOPT = ocamlopt -I +compiler-libs -g
-OCAMLDEP = ocamldep
+OCAMLC = ocamlfind ocamlc -I +compiler-libs
+OCAMLOPT = ocamlfind ocamlopt -I +compiler-libs -g
+OCAMLDEP = ocamlfind ocamldep
 OCAMLLEX = ocamllex
 #CAMLP4O =  camlp4 pa_o.cmo pa_op.cmo pr_o.cmo --
 #CAMLP4O =  camlp4 pa_o.cmo pa_op.cmo pr_dump.cmo --
@@ -38,10 +38,10 @@ DYNLOAD_OBJECTS  = fl_dynload.cmo
 DYNLOAD_XOBJECTS = $(DYNLOAD_OBJECTS:.cmo=.cmx)


-all: ocamlfind$(EXEC_SUFFIX) findlib.cma findlib_top.cma topfind $(NUMTOP) \
+all: findlib.cma findlib_top.cma topfind $(NUMTOP) \
      findlib_dynload.cma

-opt: ocamlfind_opt$(EXEC_SUFFIX) findlib.cmxa findlib_top.cmxa topfind \
+opt: findlib.cmxa findlib_top.cmxa topfind \
      findlib_dynload.cmxa

 num-top: num_top.cma
@@ -123,8 +123,6 @@ install: all
 	test $(INSTALL_TOPFIND) -eq 0 || cp topfind "$(prefix)$(OCAML_CORE_STDLIB)"
 	files=`$(SH) $(TOP)/tools/collect_files $(TOP)/Makefile.config findlib.cmi findlib.mli findlib.cma findlib.cmxa findlib$(LIB_SUFFIX) findlib.cmxs topfind.cmi topfind.mli fl_package_base.mli fl_package_base.cmi fl_metascanner.mli fl_metascanner.cmi fl_metatoken.cmi findlib_top.cma findlib_top.cmxa findlib_top$(LIB_SUFFIX) findlib_top.cmxs findlib_dynload.cma findlib_dynload.cmxa findlib_dynload$(LIB_SUFFIX) findlib_dynload.cmxs fl_dynload.mli fl_dynload.cmi META` && \
 	cp $$files "$(prefix)$(OCAML_SITELIB)/$(NAME)"
-	f="ocamlfind$(EXEC_SUFFIX)"; { test -f ocamlfind_opt$(EXEC_SUFFIX) && f="ocamlfind_opt$(EXEC_SUFFIX)"; }; \
-	cp $$f "$(prefix)$(OCAMLFIND_BIN)/ocamlfind$(EXEC_SUFFIX)"
 	# the following "if" block is only needed for 4.00beta2
 	if [ $(OCAML_REMOVE_DIRECTORY) -eq 0 -a -f "$(OCAML_CORE_STDLIB)/compiler-libs/topdirs.cmi" ]; then \
 	    cd "$(OCAML_CORE_STDLIB)/compiler-libs/"; \
